tests/unit/lib/parameter_observers/ParameterObserverCV_unittest.cc

```diff
@@ -0,0 +1,172 @@
+/*
+* BSD 3-Clause License
+*
+* Copyright (c) 2017, Shogun-Toolbox e.V. <shogun-team@shogun-toolbox.org>
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* * Redistributions of source code must retain the above copyright notice, this
+*   list of conditions and the following disclaimer.
+*
+* * Redistributions in binary form must reproduce the above copyright notice,
+*   this list of conditions and the following disclaimer in the documentation
+*   and/or other materials provided with the distribution.
+*
+* * Neither the name of the copyright holder nor the names of its
+*   contributors may be used to endorse or promote products derived from
+*   this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+* Written (W) 2017 Giovanni De Toni
+*
+*/
+#include <shogun/base/init.h>
+#include <shogun/evaluation/CrossValidation.h>
+#include <shogun/evaluation/CrossValidationSplitting.h>
+#include <shogun/evaluation/MeanSquaredError.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/LinearKernel.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/parameter_observers/ParameterObserverCV.h>
+#include <shogun/regression/KernelRidgeRegression.h>
+
+#include <gtest/gtest.h>
+#include <memory>
+
+using namespace shogun;
+
+/**
+ * This test was inspired by the meta example
```

Comment:
What do you mean here? :P Should I keep only the test and remove the meta example? Or it's better to port the meta example and keep also the test?